src/interfaces/vault.ts

```diff
@@ -304,6 +306,125 @@ export class VaultInterface<T extends ChainId> extends ServiceInterface<T> {
+   * @param vaultAddress
+   * @param tokenAddress
+   * @param amount
+   * @param overrides
+   * @returns TransactionResponse
+   */
+  async approveDeposit(
+    accountAddress: Address,
+    vaultAddress: Address,
+    tokenAddress: Address,
+    amount?: Integer,
+    overrides?: CallOverrides
+  ): Promise<TransactionResponse> {
+    const spenderAddress = await this.getDepositSpenderAddress(vaultAddress, tokenAddress);
+    return this.yearn.tokens.approve(
+      accountAddress,
+      tokenAddress,
+      spenderAddress,
+      amount ?? MaxUint256.toString(),
+      overrides
+    );
+  }
+
+  /**
+   * Approve the token amount to allow to be used for withdraw
+   * @param accountAddress
+   * @param vaultAddress
+   * @param tokenAddress
+   * @param amount
+   * @param overrides
+   * @returns TransactionResponse
+   */
+  async approveWithdraw(
+    accountAddress: Address,
+    vaultAddress: Address,
+    tokenAddress: Address,
+    amount?: Integer,
+    overrides?: CallOverrides
+  ): Promise<TransactionResponse> {
+    const spenderAddress = await this.getWithdrawSpenderAddress(vaultAddress, tokenAddress);
+    return this.yearn.tokens.approve(
+      accountAddress,
+      vaultAddress,
+      spenderAddress,
+      amount ?? MaxUint256.toString(),
+      overrides
+    );
+  }
+
+  private async getDepositSpenderAddress(vaultAddress: Address, tokenAddress: Address): Promise<Address> {
+    if (isNativeToken(tokenAddress)) return vaultAddress;
+
+    const willDepositUnderlyingToken = await this.isUnderlyingToken(vaultAddress, tokenAddress);
+    const shouldUsePartnerService = this.shouldUsePartnerService(vaultAddress);
+
+    if (willDepositUnderlyingToken && shouldUsePartnerService) {
+      const partnerTrackingContractAddress = await this.yearn.services.partner?.address;
+      if (!partnerTrackingContractAddress) throw new SdkError("Partner Tracking Contract Address not defined");
+      return partnerTrackingContractAddress;
+    }
+
+    if (willDepositUnderlyingToken && !shouldUsePartnerService) {
+      return vaultAddress;
+    }
+
+    const willZapToPickleJar = PickleJars.includes(vaultAddress);
+    const zapContract = willZapToPickleJar ? ContractAddressId.pickleZapIn : ContractAddressId.zapperZapIn;
+    const zapContractAddress = await this.yearn.addressProvider.addressById(zapContract);
+    return zapContractAddress;
+  }
+
+  private async getWithdrawSpenderAddress(vaultAddress: Address, tokenAddress: Address): Promise<Address> {
+    const willWithdrawToUnderlyingToken = await this.isUnderlyingToken(vaultAddress, tokenAddress);
+    if (willWithdrawToUnderlyingToken) return vaultAddress;
+
+    const zapContractAddress = await this.yearn.addressProvider.addressById(ContractAddressId.zapperZapOut);
+    return zapContractAddress;
+  }
+
+  private async isUnderlyingToken(vaultAddress: Address, tokenAddress: Address): Promise<boolean> {
+    const [vault] = await this.getStatic([vaultAddress]);
```

Comment:
This is why we should have a method to get a single address :) 

Looks nice though, but it's very confusing. Ideally I'd prefer something like:

```suggestion
    const vault = await this.getOneStatic(vaultAddress);
```

Reply:
Its matter of preference, since also maintaining both adds a little overhead, but do agree it can be ideal.